### ISSUE TITLE
ComplexMultiply redundant operations

### ISSUE THREAD

**Issue by ewanbarr** [arrayfire/arrayfire#1679]
In the fftconvolve.hpp (and perhaps elsewhere) for CUDA (arrayfire/src/backend/cuda/kernel/fftconvolve.hpp) complex multiplies are performed like so:

```
        out.ptr[ridx2].x = c1.x*c2.x - c1.y*c2.y;
        out.ptr[ridx2].y = (c1.x+c1.y) * (c2.x+c2.y) - c1.x*c2.x - c1.y*c2.y;
```

Is there a reason for this form for the multiply? It would appear to have several redundant operations that may reduce performance and do increase floating point propagation error.

The multiplies should be:

```
        out.ptr[ridx2].x = c1.x*c2.x - c1.y*c2.y;
        out.ptr[ridx2].y = c1.x*c2.y + c1.y*c2.x;
```

**Comment by 9prady9** [comment]
@ewanbarr Your observation is correct and thank you for reporting it.

**Comment by 9prady9** [comment]
@ewanbarr Fix for this should be pushed with 3.4.2 release.
